pkg/resource/deploy/source_eval.go

```diff
@@ -1008,6 +1021,76 @@ func inheritFromParent(child resource.Goal, parent resource.Goal) *resource.Goal
 	return &goal
 }
 
+type sourcePositions struct {
+	projectRoot string
+}
+
+func newSourcePositions(projectRoot string) *sourcePositions {
+	if projectRoot == "" {
```

Comment:
Can we be strict and just always assert that projectRoot is an abs path here?

Reply:
Possibly. I was primarily concerned with tests that don't set a project root, but this was admittedly a bit of defensive coding. I'll see if we can just make the assertion.

Reply:
Yeah, tests don't set a project root. We can either change the tests to do so or leave this code as-is.

Reply:
Leave as is for now then